Formula/httpd.rb

```diff
@@ -0,0 +1,132 @@
+    sha256 "e6ae5c4d38b40bcf3d85877ac117a1ea048de774edec3c2b0ff714bc01a6d40a" => :yosemite
+  end
+
+  skip_clean :la
+
+  deprecated_option "with-http2" => "with-nghttp2"
+
+  depends_on "openssl"
+  depends_on "pcre"
+  depends_on "apr-util"
+  depends_on "nghttp2" => :optional
+
+  if build.with?("mpm-worker") && build.with?("mpm-event")
+    raise "Cannot build with both worker and event MPMs, choose one"
+  end
+
+  def install
+    # point config files to opt_prefix instead of the version-specific prefix
+    inreplace "Makefile.in",
+      '#@@ServerRoot@@#$(prefix)#', '#@@ServerRoot@@'"##{opt_prefix}#"
+
+    # fix non-executable files in sbin dir (for brew audit)
+    inreplace "support/Makefile.in",
+      "$(DESTDIR)$(sbindir)/envvars", "$(DESTDIR)$(sysconfdir)/envvars"
+    inreplace "support/Makefile.in",
+      "envvars-std $(DESTDIR)$(sbindir);", "envvars-std $(DESTDIR)$(sysconfdir);"
+    inreplace "support/apachectl.in",
+      "@exp_sbindir@/envvars", "#{etc}/apache2/2.4/envvars"
+
+    # install custom layout
+    (buildpath/"config.layout").unlink
+    (buildpath/"config.layout").write <<-EOS.undent
+      <Layout Homebrew>
+          prefix:        #{prefix}
+          exec_prefix:   ${prefix}
+          bindir:        ${exec_prefix}/bin
+          sbindir:       ${exec_prefix}/bin
+          libdir:        ${exec_prefix}/lib
+          libexecdir:    ${exec_prefix}/libexec
+          mandir:        #{man}
+          sysconfdir:    #{etc}/apache2/2.4
+          datadir:       #{var}/www
+          installbuilddir: ${prefix}/build
+          errordir:      ${datadir}/error
+          iconsdir:      ${datadir}/icons
+          htdocsdir:     ${datadir}/htdocs
+          manualdir:     ${datadir}/manual
+          cgidir:        #{var}/apache2/cgi-bin
+          includedir:    ${prefix}/include/httpd
+          localstatedir: #{var}/apache2
+          runtimedir:    #{var}/run/apache2
+          logfiledir:    #{var}/log/apache2
+          proxycachedir: ${localstatedir}/proxy
+      </Layout>
+    EOS
+
+    args = %W[
+      --enable-layout=Homebrew
+      --enable-mods-shared=all
+      --enable-unique-id
+      --enable-ssl
+      --enable-dav
+      --enable-cache
+      --enable-logio
+      --enable-deflate
+      --enable-cgi
+      --enable-cgid
+      --enable-suexec
+      --enable-rewrite
+      --with-apr=#{Formula["apr"].opt_prefix}
+      --with-apr-util=#{Formula["apr-util"].opt_prefix}
+      --with-pcre=#{Formula["pcre"].opt_prefix}
+      --with-ssl=#{Formula["openssl"].opt_prefix}
+      --with-z=#{Formula["zlib"].opt_prefix}
+      --with-port=8080
+      --with-sslport=8443
+      --with-mpm=event
+    ]
+
+    if build.with? "nghttp2"
+      args << "--enable-http2"
+      args << "--with-nghttp2=#{Formula["nghttp2"].opt_prefix}"
+    end
+
+    (etc/"apache2/2.4").mkpath
+
+    system "./configure", *args
+    system "make"
+    system "make", "install"
+
+    (var/"apache2/log").mkpath
```

Comment:
should we do these in `post_install` as well as `etc/"apache2/2.4"`?

Reply:
and the touches below

Reply:
Do they need to be in `post_install`? Shouldn't do.

Reply:
We generally do _all_ mkpath for etc and var in post_install for $reasons that I've never quite fully understood. CC @DomT4

Reply:
Mike has explained this to be more, so, if I claimed something he would be the source :smile:. IIRC it was something to do with bottles. Possibly made redundant by https://github.com/Homebrew/brew/commit/f951a22beacdae7dd95c94eba67236fe221a3ca0?

Reply:
They are required for creating empty directories but not non-empty ones.

Reply:
Right, that makes sense, and explains why I lean on `h2o` as an internal reference.

Reply:
So that means the `(var/"apache2/run").mkpath` must be in post install, no?

Reply:
That'd be my reading of this formula as-is.